### PR TITLE
64 unittest ci

### LINKED ISSUE #64: Build unittests in CI
Our Jenkins doesn't build (let alone run...) our unittests.

It did go unnoticed, that they didn't survive the jsoncons upgrade and repo reorganisation, see #63 

At the bare minimum we should build unittests during CI builds

### PR DESCRIPTION
**Pull request by wenwenchenbosch**
close #64 

@SebastianSchildt We still need to fix some testcases. I disabled executing test cases in jenkins

### CODE REVIEW

**Comment by wenwenchenbosch** [comment]
pls merge https://github.com/eclipse/kuksa.val/pull/63 at first

**Comment by SebastianSchildt** [comment]
#63 is merged, but this looks weird in the diff. Can you rebase it? https://github.com/boschresearch/kuksa.val/tree/64_unittest_ci is "6 ahead, 1 behind", that does not seem right

File: test/unit-test/readme.md [hunk 1]
```diff
@@ -6,17 +6,21 @@ Copy the vss signal tree json manually from [here](https://github.com/GENIVI/veh
...
+cmake .. -DBUILD_UNIT_TEST=ON
 
 make
 
-
+```
 you can execute using
 
+```
 ./w3c-unit-test
```
Comment (SebastianSchildt) [inline test/unit-test/readme.md:23]: Should we rename this to kuksa.val-unit-test or something similar?

**Review by SebastianSchildt — APPROVED** [review]
Ok looks fine now. Should be merged to fix build and see breakage there. We can repair unittests in a later pr.
No new dependencies.

File: Jenkinsfile [hunk 2]
```diff
@@ -37,4 +37,12 @@ node('docker') {
             archiveArtifacts artifacts: 'artifacts/*.xz' 
         }
     }
+    docker.image('kuksa-val-dev:ubuntu20.04').inside("-v /var/run/docker.sock:/var/run/docker.sock"){ 
+        stage('Test') {
+            sh '''
+                cd /kuksa.val/build
+                # TODO and you may need sudo right for testing ctest --build-config Debug --output-on-failure --parallel 8
```
Comment (eriksven) [inline Jenkinsfile:44]: was your intention to resolve this TODO in later PR?

Comment (wenwenchenbosch) [reply Jenkinsfile:44]: I think so. This PR let test cases building automatically. And the test cases will be also executed, after all test cases show positive.

**Pull request merged**